src/_pytest/assertion/util.py

```diff
@@ -354,7 +355,7 @@ def _compare_eq_iterable(
 def _compare_eq_sequence(
     left: Sequence[Any],
     right: Sequence[Any],
-    highlighter: _HighlightFunc,
+    highlighter: Union[_HighlightFunc, Callable[[str], str]],
```

Comment:
Wasn't too sure if this was the best way to fix the type error, or if I should make a dummy _HighlightFunc for when you don't want to actually use the highlighter